tokio/src/signal/unix.rs

```diff
@@ -391,13 +399,17 @@ pub fn signal(kind: SignalKind) -> io::Result<Signal> {
 }
 
 impl Signal {
-    #[doc(hidden)] // TODO: Dox
+    /// Receive the next signal notification event.
+    ///
+    /// `None` is returned if no more events can be received by this stream.
```

Comment:
Anyway I could convince you to add an example? :) You can use `no_run` to avoid having it run in tests.

Reply:
Oh yeah I can add these in!